src/abTestingInfra/emitter.ts

```diff
@@ -0,0 +1,72 @@
+import EventEmitter from 'eventemitter3';
+
+const emitter = new EventEmitter();
+
+export const EXPERIMENT_PLAYED = 'EXPERIMENT_PLAYING';
+export const EXPERIMENT_WON = 'EXPERIMENT_WON';
+
+declare global {
+  interface Window {
+    experiments: Map<string, boolean>;
+  }
+}
+
+if (typeof window !== 'undefined') {
+  window.experiments = new Map<string, boolean>();
+}
```

Comment:
Why do we need `window` here? Could this be a map that's private to the file?

Reply:
> Last bit of review! Lmk if you'd like to pair on the `permutations.ts` cleanup; I think this is an opportunity where working together will speed us up.
> 
> Otherwise, high-level comments include:
> 
> * Since we're compiling the code anyway, should we be using TypeScript for the `experimentExtractor` files and running them through babel as a `prepublish` step?

Done!

> * Please ensure ESLint is installed, and is using the same config as the other project(s)

Done!